app/angular/src/client/preview/angular/ComponentClassFromStoryComponent.ts

```diff
@@ -0,0 +1,146 @@
+  getComponentDecoratorMetadata,
+  getComponentInputsOutputs,
+} from './NgComponentAnalyzer';
+import { RenderNgAppService } from './RenderNgAppService';
+
+const distinctComponentInputsOutputsWithProps = (
+  ngComponentInputsOutputs: ComponentInputsOutputs,
+  props: ICollection = {}
+) => {
+  const inputs = ngComponentInputsOutputs.inputs
+    .filter((i) => i.templateName in props)
+    .map((i) => i.templateName);
+  const outputs = ngComponentInputsOutputs.outputs
+    .filter((o) => o.templateName in props)
+    .map((i) => i.templateName);
+  return {
+    inputs,
+    outputs,
+    otherProps: Object.keys(props).filter((k) => ![...inputs, ...outputs].includes(k)),
+  };
+};
+
+/**
+ * Wraps the story component into a component
+ *
+ * @param component
+ * @param initialProps
+ */
+export const createComponentClassFromStoryComponent = (
+  component: any,
+  initialProps?: ICollection
+): Type<any> => {
+  const ngComponentMetadata = getComponentDecoratorMetadata(component);
+  const ngComponentInputsOutputs = getComponentInputsOutputs(component);
+
+  const {
+    inputs: initialInputs,
+    outputs: initialOutputs,
+    otherProps: initialOtherProps,
+  } = distinctComponentInputsOutputsWithProps(ngComponentInputsOutputs, initialProps);
+
+  const templateInputs = initialInputs.map((i) => `[${i}]="${i}"`).join(' ');
+  const templateOutputs = initialOutputs.map((i) => `(${i})="${i}($event)"`).join(' ');
+
+  @Component({
+    selector: RenderNgAppService.SELECTOR_STORYBOOK_WRAPPER,
+    // Simulates the `component` integration in a template
+    // `props` are converted into Inputs/Outputs to be added directly in the template so as the component can use them during its initailization
+    // - The outputs are connected only once here
+    // - Only inputs present in initial `props` value are added. They will be overwritten and completed as necessary after the component is initialized
+    template: `<${ngComponentMetadata.selector} ${templateInputs} ${templateOutputs} #storyComponentRef></${ngComponentMetadata.selector}>`,
+  })
+  class StoryBookComponentWrapperComponent implements AfterViewInit, OnDestroy {
+    private storyPropsSubscription: Subscription;
+
+    @ViewChild('storyComponentRef', { static: true }) storyComponentElementRef: ElementRef;
+
+    @ViewChild('storyComponentRef', { read: ViewContainerRef, static: true })
+    storyComponentViewContainerRef: ViewContainerRef;
+
+    constructor(
+      @Inject(STORY_PROPS) private storyProps$: Subject<ICollection | undefined>,
+      private changeDetectorRef: ChangeDetectorRef
+    ) {
+      // Initializes template Inputs/Outputs values
+      Object.assign(this, initialProps);
+    }
+
+    ngAfterViewInit(): void {
+      // Initializes properties that are not Inputs | Outputs
+      // Allows story props to override local component properties
+      initialOtherProps.forEach((p) => {
+        (this.storyComponentElementRef as any)[p] = initialProps[p];
+      });
```

Comment:
If I correctly followed all your comments in this PR (and in another issue), you're suggesting to only assign `initialInputs` and `initialOutputs`, and not the whole `initialProps` which can lead to setting values to "private/internal" values? 

In my opinion, it makes perfect sense and should work like that to better reflect the real-life use of the component. However, it's a breaking change so we can not introduce such behavior until 7.0.

Reply:
In this PR for the moment it works as before. 
But yes, I suggest that this is no longer the default behavior. 
I think, it can be the subject of a dedicated PR or we could first add a warning message.
Then add a flag (like `overrideInternalComponentProperty`) to make it optional. 🤷‍♂️